src/main/java/core/StandardForwardModel.java

```diff
@@ -116,7 +116,7 @@ public void endPlayerTurn(AbstractGameState gs) {
      * @param gs                     - current game state.
      * @param firstPlayerOfNextRound the first player to act in the next round
      */
-    public final void endRound(AbstractGameState gs, int firstPlayerOfNextRound) {
+    public void endRound(AbstractGameState gs, int firstPlayerOfNextRound) {
```

Comment:
I think we should keep endRound() as final. This is because it has the code that published END_ROUND events, which otherwise people will forget to include (this is why we made it final in the great TurnOrder refactor).

The ForwardModel can and should implement a specific set of end Round functionality, part of which is to call this method.

Reply:
But this is the same as endPlayerTurn, which creates the TURN_OVER events, and is similarly not final. Feeling strongly on making them all final? (People can also always forget to call the super methods in their own implementation anyway, maybe documentation is enough to prevent this - in which case allowing non-final methods would be better I think, as you'd have more direct access to the documentation for that method you're trying to write that reminds you to do all the right functionality in addition to your own)

Reply:
Yes. I agree. Both should be final!